src/ai/defaultai.cc

```diff
@@ -3392,26 +3392,16 @@ void DefaultAI::diplomacy_actions(const Time& gametime) {
 			continue;
 		}
 		if (player_statistics.get_diplo_score(opn) >= 35) {
-			if (((other_player->team_number() == 0 ||
-			      player_statistics.team_power(me->team_number()) >
-			         player_statistics.team_power(other_player->team_number())) &&
-			     player_statistics.members_in_team(me->team_number()) <
-			        player_statistics.players_active() - 1) ||
-			    (player_statistics.members_in_team(other_player->team_number()) ==
-			        player_statistics.players_active() - 1 &&
-			     me->team_number() != other_player->team_number())) {
+			Widelands::DiplomacyAction action = player_statistics.join_or_invite(opn, gametime);
```

Comment:
as we return the diplo action we could directly use it in the send_player diplomacy command making the if loop here superfluous. Logging could be done in the help function

Reply:
OK. I almost did it, but it complicated the return code here, and we need to check the return value anyway. But using it directly was part of the original plan.

Reply:
OTOH it's cleaner to have the logging in the same place as the actual execution… I'll bring that too, and then we don't even have to fiddle with reporting skipped actions to the caller. Is that OK with you?

Reply:
as far as I understand we could use exactly the return codes already provided as a kRefuseJoin without a Join request gets simply ignored.

Reply:
> OTOH it's cleaner to have the logging in the same place as the actual execution… I'll bring that too, and then we don't even have to fiddle with reporting skipped actions to the caller. Is that OK with you?

yes please. Please note I made some changes to the original branch.